### ISSUE TITLE
Noetic networking netstat VS NetworkManager (round one, fight)

### ISSUE THREAD

**Issue by JanezCim** [UbiquityRobotics/magni_robot#185]
Image: 2021-08-06-focal
Rpi4 

**TLDR:**
I think i cant set a static ip in the RPI with `netstat` on `eth0` because `pifi` is installed

**Long version:**
While tried setting the static IP on eth0 for lidar to work, we've come across a problem, that we cant configure it with the netstat networking tool, which is default on ubuntu 20.04. These tutorials were followed:
https://pimylifeup.com/ubuntu-20-04-static-ip-address/
https://www.linkedin.com/pulse/installing-ubuntu-server-2004-lts-raspberry-pi4-erik-larson/
https://discourse.ubuntu.com/t/how-to-install-ubuntu-server-on-your-raspberry-pi/14660

Since pifi is pre-configured on the RPI images (and afaik pifi uses networkManager from net-tools), I suspect that netstat somehow gets over-written with the NetworkManager configuration which prevented any of the above tutorials to work. Now I have been able to overcome this problem by simply installing `sudo apt install net-tools` and then setting the static ip from `/etc/network/interfaces`, like in the good'ol times, but this is really a hacky solution since net-tools has not been mainained for some time now.

**The question now is: How do we make netstat configuration work together with pifi so we can have a long term solution to this problem?**

**Comment by MoffKalast** [comment]
There's definitely something very weird or wrong with the current netplan install, looking at what's standard:
- there is supposedly some cloud init in /etc/cloud/... which does not exist at all
- the /etc/netplan dir is empty, it should already contain some yaml configs by default
- looking at the desktop settings gui, the Wired option is completely missing in the Network tab where it ought to be, if you switch off wifi and try to go with a dhcp router connection it'll show up as "wired unmanaged" in the taskbar, implying there's nothing in control of it


Adding the ethernet device via nmcli has no effect whatsoever. Not sure what I was expecting, but yeah.

I'm not sure if pifi can be blamed for all that, it only sends NM the odd add network command.

**Comment by rohbotics** [comment]
The noetic image is configured to use systemd-networkd not netplan. Because NetworkManager sees that systemd-networkd is managing the interface, the GUI is showing it as unmanaged.

Can we try something like this: https://wiki.archlinux.org/title/systemd-networkd#Wired_adapter_using_a_static_IP

**Comment by rohbotics** [comment]
Also here is where we set the current config: https://github.com/UbiquityRobotics/pi_image2/blob/master/build_image.py#L194

**Comment by JanezCim** [comment]
I've now taken a nother look into this and i've managed to set a static ip with networkd by

`sudo nano /etc/systemd/network/10-eth-dhcp.network`

and replacing everything with

```
[Match]
Name=eth*

[Network]
Address=192.168.42.125/24
```

and then `sudo reboot` or `sudo systemctl restart systemd-networkd`

I havent tested this with lidar yet though because i dont have one, could you try it out if it works?

**Comment by anfederman** [comment]
This is already in breadcrumb learn documentation. Please note the default
lie at is set to the replicate ld06.

**Comment by JanezCim** [comment]
okay cool! @anfederman can you confirm this network config works with ls lidar?

**Comment by anfederman** [comment]
Pifi and lslidar work once netstat  set. I even had Rviz showing lidar  +
sonar,

Interesting spell corrections:

megastar = netstat
lslidar = pelican

**Comment by JanezCim** [comment]
can you elaborate what means that you have netstat set. Best would be if you give us a set of commands and eddits that made it work for you.

**Comment by anfederman** [comment]
The Feb dev image is setup for working with the ld06 lidar.

To get the LS lidar to work you need to edit the Magni bring up  to set
ldlidar  false, LS lidar  true, and also  use either Magellan or megastar
to set the Ethernet IP  address for the robot. The instructions for this
are in breadcrumb learn.

**Comment by JanezCim** [comment]
Today i've used this setup to setup LS lidar: https://github.com/UbiquityRobotics/magni_robot/issues/185#issuecomment-1036157901 and this has also been similarly solved in breadcrumb repo.
And it seems to be working normally. I think this solves this issue. reopen if necesseary
